openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -159,15 +162,28 @@ def get_serializer_class(self):
     def create(self, request, *args, **kwargs):
         if request.data.get('profile'):
             request.data['profile']['organization'] = self.request.user.profile.organization_id
-        password = uuid4().hex[:8]
-        request.data['password'] = password
-        user_serializer = UserSerializer(data=request.data)
-        if user_serializer.is_valid():
-            user = user_serializer.save()
-            send_registration_email(user, password, request.scheme)
-            return Response(UserSerializer(user).data, status=status.HTTP_201_CREATED)
 
-        return Response({**user_serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+        is_created, res_data = create_user(request.data, request.scheme)
+        if is_created:
+            return Response(UserSerializer(res_data).data, status=status.HTTP_201_CREATED)
+
+        return Response(res_data, status=status.HTTP_400_BAD_REQUEST)
+
+    def bulk_registration(self, request, *args, **kwargs):
+        if not request.FILES.get('file'):
+            return Response('File is required!', status=status.HTTP_400_BAD_REQUEST)
+
+        file_data = io.StringIO(request.FILES['file'].read().decode('utf-8'))
+        file_reader = csv.DictReader(file_data)
+
+        required_col_names = {'name', 'username', 'email', 'organization_id', 'role', 'employee_id', 'language'}
+        if not set(file_reader.fieldnames) == required_col_names:
+            return Response('Invalid file format! Column names are incorrect!', status=status.HTTP_400_BAD_REQUEST)
+
+        users = get_user_data_from_bulk_registration_file(file_reader, self.request.user.profile.organization_id)
```

Comment:
Should we introduce a util method to get current user org, we've 3 places where we are using the same line of code?
`get_request_user_org(self.request)`